epcis/src/main/java/io/openepcis/model/dto/EPCISEventResponse.java

```diff
@@ -50,6 +51,21 @@ public class EPCISEventResponse {
   public EPCISEventResponse(final EPCISEvent epcisEvent) {
     this.epcisEvent = epcisEvent;
     this.contextInfo = getContextInfoFromEvent(epcisEvent);
+
+    // Populating the namespaces directly from context during xml query
+    if (!contextInfo.isEmpty()) {
+      final DefaultJsonSchemaNamespaceURIResolver namespaceURIResolver =
+          DefaultJsonSchemaNamespaceURIResolver.getContext();
+      namespaceURIResolver.resetEventNamespaces();
+
+      for (Object item : contextInfo) {
+        if (item instanceof Map<?, ?>) {
+          final Map<String, String> namespaces = (Map<String, String>) item;
+          namespaces.forEach(
+              (key, value) -> namespaceURIResolver.populateEventNamespaces(value, key));
+        }
+      }
+    }
```

Comment:
duplicated same code in EPCISQueryDocument
please try to come up with one single generic implemenation